src/types.ts

```diff
@@ -80,3 +80,40 @@ export interface BehavioralQuestion {
 export interface DetailedResponse {
   detail: string;
 }
+
+export interface RawInterviewData {
+  interview_id: string;
+  date_effective: string;
+  date_completed: string;
+  interviewer: number;
+  interviewee: number;
+  status: Status;
+  technical_question: TechnicalQuestion;
+  behavioral_questions: BehavioralQuestion[];
+}
+
+export interface RawInterviewAvailabilityData {
+  user_id: number;
+  availability: boolean[][];
+}
+
+export interface RawMemberData {
```

Comment:
Are these fields all guaranteed? I think some can be undefined so you might want to put them as field?: string. 

Also, this pattern might be nice:
```
interface Member {
  id: number; // user id
  username: string;
  created: string; // ISO 8601 date string
  email: string;
  role: string;
  firstName: string;
  lastName: string;
  preview?: string;
  major?: string;
  gradDate?: string; // ISO 8601 date string or null
  discordUsername: string;
  linkedin?: SocialField;
  github?: SocialField;
  leetcode?: SocialField;
  resumeUrl?: string;
  local?: string;
  bio?: string;
  discordId: number;
}

type RawMemberData = Omit<BaseMember, 'firstName' | 'lastName' | 'gradDate' | 'discordUsername' | 'resumeUrl' | 'discordId'> & {
  user: number;
  first_name: string;
  last_name: string;
  grad_date?: string;
  discord_username: string;
  resume_url?: string;
  discord_id: number;
};
```

also, you can set the dates to `DateTime` types if that makes things easier instead of string.

Reply:
good call on optionals. I think for now I'm going to keep it simple and just write out the explicit mapping between names, but we can move to using Omit if it ends up growing too large

Reply:
Also will add date parsing to deserializers when I consolidate them outside of services module